PWGHF/HFL/TableProducer/electronSelectionWithTpcEmcal.cxx

```diff
@@ -365,18 +403,86 @@ struct HfElectronSelectionWithTpcEmcal {
-                    eMatchEmcCluster, etaMatchEmcCluster, phiMatchEmcCluster, m02MatchEmcCluster, m20MatchEmcCluster, cellEmcCluster, timeEmcCluster, deltaEtaMatch, deltaPhiMatch, isEMcal);
+        /////////////////////////////////////////////////////////////            photonic electron          ///////////////////////////////////////////////////////
+
+        isLSElectronFound = 0;
+        isULSElectronFound = 0;
+        bool isLSElectron = 0;
+        bool isULSElectron = 0;
+        float invMassElectron = 0.;
+        float massLike = 0;
+        float massUnLike = 0;
+
+        for (const auto& pTrack : tracks) {
+
+          if (pTrack.globalIndex() == matchTrack.globalIndex())
+            continue;
+
+          // Apply Associated electron Selection
+
+          if (!selAssoTracks(pTrack)) {
+            continue;
+          }
+          Int_t PDGe1 = kElectron;
+          Int_t PDGe2 = kElectron;
+          if (matchTrack.sign() > 0)
+            PDGe1 = kPositron;
+          if (pTrack.sign() > 0)
+            PDGe2 = kPositron;
+          KFPTrack kfpTrack = createKFPTrackFromTrack(matchTrack);
+          KFPTrack kfpAssociatedTrack = createKFPTrackFromTrack(pTrack);
+          KFParticle KFTrack(kfpTrack, PDGe1);
+          KFParticle kfAssociatedTrack(kfpAssociatedTrack, PDGe2);
+          const KFParticle* LcDaughters[2] = {&KFTrack, &kfAssociatedTrack};
+          KFNonHfe.SetConstructMethod(2);
+          KFNonHfe.Construct(LcDaughters, 2);
+
+          Int_t ndf = KFNonHfe.GetNDF();
+          Double_t chi2recg = KFNonHfe.GetChi2() / ndf;
+          if (ndf < 1.0) {
+            continue;
+          }
+
+          if (std::sqrt(std::abs(chi2recg)) > chiSquareMax) {
+            continue;
+          }
+
+          invMassElectron = RecoDecay::m(std::array{pTrack.pVector(), matchTrack.pVector()}, std::array{MassElectron, MassElectron});
+
+          // for like charge
+          if (pTrack.sign() == matchTrack.sign()) {
+            massLike = invMassElectron;
+            isLSElectron = true;
+            registry.fill(HIST("hLikeMass"), massLike);
+          }
+          // for unlike charge
+          if (pTrack.sign() != matchTrack.sign()) {
+            massUnLike = invMassElectron;
+            isULSElectron = true;
+            registry.fill(HIST("hUnLikeMass"), massUnLike);
+          }
+
+          // for like charge
+          if (isLSElectron && (invMassElectron <= invariantMass)) {
+            massLike = invMassElectron;
+            isLSElectronFound = 1;
+          }
+          // for unlike charge
+          if (isULSElectron && (invMassElectron <= invariantMass)) {
+            massUnLike = invMassElectron;
+            isULSElectronFound = 1;
+          }
+        }
+        electronSel(track.collisionId(), matchTrack.globalIndex(), etaMatchTrack, phiMatchTrack, ptMatchTrack, pMatchTrack, trackRapidity, matchTrack.dcaXY(), matchTrack.dcaZ(), matchTrack.tpcNSigmaEl(), matchTrack.tofNSigmaEl(),
```

Comment:
this would fill only one (the first) ULS and LS pair information right? It does not fill all information. You will need all pair combinations.

Reply:
It fills only one ULS and LS pair's information because we aim to identify which electron forms a Like Sign (LS) pair and which forms an Unlike Sign (ULS) pair. Additionally, I fill histograms for the invariant mass of LS and ULS pairs. If any additional histograms are required, we can create them.